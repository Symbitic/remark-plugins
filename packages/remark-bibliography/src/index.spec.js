/* eslint-env jest */
import bibliography from './index'
import html from 'rehype-stringify'
import markdown from 'remark-parse'
import path from 'path'
import remark2rehype from 'remark-rehype'
import unified from 'unified'

const ASSETS = path.resolve(__dirname, '..', 'examples')

const parse = (bib, str) =>
  unified()
    .use(markdown)
    .use(() => (tree, file) => {
      file.history = [ path.join(ASSETS, 'examples.md') ]
      file.data = { bibliography: bib }
    })
    .use(bibliography)
    .use(remark2rehype)
    .use(html)
    .process(str)
    .then(data => data.toString())

const EXAMPLE1 = `This is a citation: (@singh2016).`

const tests = [
  [ 'process a basic bibliography', 'example.json', EXAMPLE1 ]
]

describe.each(tests)('remark-bibliography', (desc, file, source) => {
  it(`should ${desc}`, () => {
    return expect(parse(file, source)).resolves.toMatchSnapshot()
  })
})
