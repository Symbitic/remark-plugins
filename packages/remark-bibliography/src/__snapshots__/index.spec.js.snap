// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark-bibliography should process a basic bibliography 1`] = `
"<p>This is a citation: <a href=\\"#ref-singh2016\\" class=\\"citation\\">(Singh 2016)</a>.</p>
<div class=\\"references\\"><h2>References</h2><ol class=\\"ref-list\\">
<li id=\\"ref-moller2008\\" class=\\"reference\\">Akenine-Moller, Thomas, Eric Haines, and Naty Hoffman. 2008. <em>Real Time Rendering, Third Edition</em>. CRC Press. http://www.realtimerendering.com/.</li>
<li id=\\"ref-gregory2014\\" class=\\"reference\\">Gregory, Jason. 2014. <em>Game Engine Architecture, Second Edition.</em> CRC Press. https://doi.org/10.1186/1758-2946-3-47.</li>
<li id=\\"ref-singh2016\\" class=\\"reference\\">Singh, Parminder. 2016. <em>Vulkan Essentials</em>. Packt Publishing. https://www.packtpub.com/application-development/vulkan-essentials.</li>
</ol></div>"
`;
